### ISSUE TITLE
OMP compiled ocean-ice experiments crash with "The linear system is singular "

### ISSUE THREAD

**Issue by nikizadehgfdl** [NOAA-GFDL/MOM6#588]
The MOM6_GOLD_SIS2 experiment that runs fine without -openmp crashes as follows when MOM6 is compiled with -openmp 

FATAL from PE    28: The linear system is singular !

```
forrtl: error (76): Abort trap signal
Image              PC                Routine            Line        Source             
fms_MOM6_SIS2_com  00000000017D6B36  mpp_mod_mp_mpp_er          50  mpp_util_mpi.inc
fms_MOM6_SIS2_com  0000000000E2BB05  regrid_solvers_mp          76  regrid_solvers.F90
fms_MOM6_SIS2_com  0000000000BAC247  regrid_edge_value         571  regrid_edge_values.F90
fms_MOM6_SIS2_com  0000000000E0E76C  mom_remapping_mp_         351  MOM_remapping.F90
fms_MOM6_SIS2_com  0000000000E0CB41  mom_remapping_mp_         178  MOM_remapping.F90
fms_MOM6_SIS2_com  0000000000796A9B  mom_diag_remap_mp         355  MOM_diag_remap.F90
fms_MOM6_SIS2_com  0000000000733A86  mom_diag_mediator         785  MOM_diag_mediator.F90
fms_MOM6_SIS2_com  000000000144F78B  ideal_age_example         526  ideal_age_example.F90
fms_MOM6_SIS2_com  0000000000C8F9D8  mom_tracer_flow_c         550  MOM_tracer_flow_control.F90
fms_MOM6_SIS2_com  0000000000B1252C  mom_diabatic_driv        1162  MOM_diabatic_driver.F90
fms_MOM6_SIS2_com  00000000007C183C  mom_mp_step_mom_t        1180  MOM.F90
fms_MOM6_SIS2_com  00000000007B4648  mom_mp_step_mom_         1024  MOM.F90
fms_MOM6_SIS2_com  0000000000699153  ocean_model_mod_m         475  ocean_model_MOM.F90
fms_MOM6_SIS2_com  000000000040C5EB  MAIN__                   1020  coupler_main.F90
```

This happens for both Intel and PGI compilers

**Comment by nikizadehgfdl** [comment]
The issue is in none of the files appearing in the above traceback! It is in 
src/parameterizations/vertical/MOM_diabatic_aux.F90

I'll make a pull request shortly.

**Comment by adcroft** [comment]
Closed by NOAA-GFDL/MOM6@a6b1c7b2980
